internal/ipmi/ipmi.go

```diff
@@ -112,11 +112,10 @@ func (i *Ipmi) PowerResetBmc(ctx context.Context, resetType string) (ok bool, er
 func (i *Ipmi) PowerOn(ctx context.Context) (status bool, err error) {
 	s, err := i.IsOn(ctx)
 	if err != nil {
-		return false, err
+		return false, errors.Wrap(err, "error checking power state")
```

Comment:
I'd suggest the dynamic error here be defined as a type `ErrCheckPowerState`

Reply:
sorry, I'm not familiar enough with the design philosophy for custom errors in bmclib to feel comfortable adding this. Also since there are no other custom errors in this package already, this would be the first and that adds to the not comfortable feeling. Maybe another PR maybe for this?

Reply:
Yep, can be done in a separate PR.
Since https://github.com/bmc-toolbox/bmclib/issues/204 updates the linter config which will catch these dynamic errors - I suggest we start defining newer errors at the top of any file they are being declared in, it'll just take off some linting work later.